src/pytorch_lightning/serve/servable_module_validator.py

```diff
@@ -0,0 +1,142 @@
+import time
+from multiprocessing import Process
+from typing import Any, Dict, Optional
+
+import requests
+import torch
+from typing_extensions import Literal
+
+import pytorch_lightning as pl
+from pytorch_lightning.callbacks import Callback
+from pytorch_lightning.serve.servable_module import ServableModule
+from pytorch_lightning.strategies import DDPFullyShardedNativeStrategy, DDPFullyShardedStrategy, DeepSpeedStrategy
+from pytorch_lightning.utilities.imports import _RequirementAvailable
+from pytorch_lightning.utilities.model_helpers import is_overridden
+from pytorch_lightning.utilities.rank_zero import rank_zero_only
+
+_NOT_SUPPORTED_STRATEGIES = (
+    DeepSpeedStrategy,
+    DDPFullyShardedNativeStrategy,
+    DDPFullyShardedStrategy,
+)
+
+
+class ServableModuleValidator(Callback):
```

Comment:
Why should this be a Callback? There is no connection to training. The sanity check could also be a simple function executing all these steps.

Is there a real benefit from doing that inside the trainer? 

it's not more boilerplate to call the checking function compared to instantiate the callback and add it to the trainer.

Reply:
I believe this is a common sentiment shared by researchers, but I don't share it. Validating Serving for me is part of the training and should be done on_train_start and possibly on on_train_end.

It might evolve to do more, so it fits better in a callbacks, especially with shared models.